pkg/controller/vitessbackupschedule/vitessbackupschedule_controller.go

```diff
@@ -0,0 +1,624 @@
+}
+
+// cleanupJobsWithLimit removes all Job objects from the cluster ordered by oldest to newest and
+// respecting the given limit, keeping minimum "limit" jobs in the cluster.
+func (r *ReconcileVitessBackupsSchedule) cleanupJobsWithLimit(ctx context.Context, jobs []*kbatch.Job, limit int32) {
+	if limit == -1 {
+		return
+	}
+
+	sort.SliceStable(jobs, func(i, j int) bool {
+		if jobs[i].Status.StartTime == nil {
+			return jobs[j].Status.StartTime != nil
+		}
+		return jobs[i].Status.StartTime.Before(jobs[j].Status.StartTime)
+	})
+
+	for i, job := range jobs {
+		if int32(i) >= int32(len(jobs))-limit {
+			break
+		}
+		if err := r.client.Delete(ctx, job, client.PropagationPolicy(metav1.DeletePropagationBackground)); (err) != nil {
+			log.WithError(err).Errorf("unable to delete old job: %s", job.Name)
+		} else {
+			log.Infof("deleted old job: %s", job.Name)
+		}
+	}
+}
+
+func (r *ReconcileVitessBackupsSchedule) removeTimeoutJobs(ctx context.Context, jobs []*kbatch.Job, vbscName string, timeout int32) error {
+	if timeout == -1 {
+		return nil
+	}
+	for _, job := range jobs {
+		jobStartTime, err := getScheduledTimeForJob(job)
+		if err != nil {
+			return err
+		}
+		if jobStartTime.Add(time.Minute * time.Duration(timeout)).Before(time.Now()) {
+			if err = r.client.Delete(ctx, job, client.PropagationPolicy(metav1.DeletePropagationBackground)); (err) != nil {
+				log.WithError(err).Errorf("unable to delete timed out job: %s", job.Name)
+			} else {
+				log.Infof("deleted timed out job: %s", job.Name)
+			}
+			timeoutJobsCount.WithLabelValues(vbscName, metrics.Result(err)).Inc()
+		}
+	}
+	return nil
+}
+
+func isJobFinished(job *kbatch.Job) (bool, kbatch.JobConditionType) {
+	for _, c := range job.Status.Conditions {
+		if (c.Type == kbatch.JobComplete || c.Type == kbatch.JobFailed) && c.Status == corev1.ConditionTrue {
+			return true, c.Type
+		}
+	}
+
+	return false, ""
+}
+
+func getScheduledTimeForJob(job *kbatch.Job) (*time.Time, error) {
+	timeRaw := job.Annotations[scheduledTimeAnnotation]
+	if len(timeRaw) == 0 {
+		return nil, nil
+	}
+
+	timeParsed, err := time.Parse(time.RFC3339, timeRaw)
+	if err != nil {
+		return nil, err
+	}
+
+	return &timeParsed, nil
+}
+
+func (r *ReconcileVitessBackupsSchedule) createJob(ctx context.Context, vbsc *planetscalev2.VitessBackupSchedule, scheduledTime time.Time) (*kbatch.Job, error) {
+	name := fmt.Sprintf("%s-%d", vbsc.Name, scheduledTime.Unix())
+
+	meta := metav1.ObjectMeta{
+		Labels: map[string]string{
+			planetscalev2.BackupScheduleLabel: vbsc.Name,
+		},
+		Annotations: make(map[string]string),
+		Name:        name,
+		Namespace:   vbsc.Namespace,
+	}
+	for k, v := range vbsc.Annotations {
+		meta.Annotations[k] = v
+	}
+
+	// Add the user-defined annotations
+	for k, v := range vbsc.Spec.Annotations {
+		meta.Annotations[k] = v
+	}
+
+	meta.Annotations[scheduledTimeAnnotation] = scheduledTime.Format(time.RFC3339)
+
+	for k, v := range vbsc.Labels {
+		meta.Labels[k] = v
+	}
```

Comment:
FWIW, you can use `maps.Copy` https://pkg.go.dev/golang.org/x/exp/maps#Copy

Reply:
That's a good point, fixed via 54ec222